org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/yaml/YAMLLanguageServer.java

```diff
@@ -13,27 +13,108 @@
+import org.eclipse.core.resources.IResourceDelta;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.FileLocator;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.Status;
 import org.eclipse.jface.preference.IPreferenceStore;
+import org.eclipse.jface.util.IPropertyChangeListener;
+import org.eclipse.jface.util.PropertyChangeEvent;
+import org.eclipse.lsp4e.LanguageServersRegistry;
+import org.eclipse.lsp4e.LanguageServersRegistry.LanguageServerDefinition;
+import org.eclipse.lsp4e.LanguageServiceAccessor;
 import org.eclipse.lsp4e.server.ProcessStreamConnectionProvider;
 import org.eclipse.lsp4j.DidChangeConfigurationParams;
+import org.eclipse.lsp4j.DidChangeWorkspaceFoldersParams;
+import org.eclipse.lsp4j.InitializeResult;
+import org.eclipse.lsp4j.WorkspaceFolder;
+import org.eclipse.lsp4j.WorkspaceFoldersChangeEvent;
 import org.eclipse.lsp4j.jsonrpc.messages.Message;
+import org.eclipse.lsp4j.jsonrpc.messages.ResponseMessage;
 import org.eclipse.lsp4j.services.LanguageServer;
 import org.eclipse.wildwebdeveloper.Activator;
 import org.eclipse.wildwebdeveloper.embedder.node.NodeJSManager;
 
 import com.google.gson.Gson;
 import com.google.gson.reflect.TypeToken;
 
+@SuppressWarnings("restriction")
 public class YAMLLanguageServer extends ProcessStreamConnectionProvider {
+	private static final String SETTINGS_KEY = "settings";
+	private static final String YAML_KEY = "yaml";
+	private static final String VALIDATE_KEY = "validate";
+	private static final String COMPLETION_KEY = "completion";
+	private static final String HOVER_KEY = "hover";
+	private static final String SCHEMAS_KEY = "schemas";
 	
-	private String cachedSchema;
+ 	private static final IPreferenceStore store = Activator.getDefault().getPreferenceStore();
+ 	private static final LanguageServerDefinition yamlLsDefinition = LanguageServersRegistry.getInstance().getDefinition("org.eclipse.wildwebdeveloper.yaml");
+ 	private static final IPropertyChangeListener psListener = new IPropertyChangeListener() {
+		@Override
+		public void propertyChange(PropertyChangeEvent event) {
+			if (YAMLPreferenceInitializer.YAML_SCHEMA_PREFERENCE.equals(event.getProperty())) {
+				Map<String, Object> yaml = getYamlSchemaOptions();
+				Map<String, Object> settings = new HashMap<>();
+				settings.put(YAML_KEY, yaml);
+				Map<String, Object> options = new HashMap<>();
+				options.put(SETTINGS_KEY, settings);
+				DidChangeConfigurationParams params = new DidChangeConfigurationParams(settings);
+				LanguageServiceAccessor.getActiveLanguageServers(null).stream().filter(server -> yamlLsDefinition.equals(LanguageServiceAccessor.resolveServerDefinition(server).get()))
+					.forEach(ls -> ls.getWorkspaceService().didChangeConfiguration(params));
+			}
+		}
+ 	};
+ 	
+ 	private static final IResourceChangeListener wsFoldersListener = new IResourceChangeListener() {
```

Comment:
This is more something to implement in LSP4E directly. But I believe before that, it would be more interesting to investigate whether the Yaml-LS couldn't just ignore the rootURI at that time.

Reply:
I didn't found this in LSP4E and I don't know if this is something to be implemented there. I see this like something that is YAML LS specific as the Workspace Folders sent to the LS are only Project folders (without their internal sub-folders) - which makes less overload in communication between client and server. 
And I believe, YAML LS can do the same by itself, if it was properly configured at start by LSP4E (see: https://github.com/eclipse/wildwebdeveloper/issues/461#issuecomment-661559744 on `rootUri`)

On why YAML LS "couldn't just ignore the riootUri" - if it will ignore that - where and how it can find the workspace root then? And it is probably has such a problem because LSP4E configures `rootUri` to point to one of the Workspace projects and not to the Workspace root (It's probably OK for VSCode, but not for Eclipse's Workspace, imho). That's why these Workspace Folders (projects) now are required by LS.